dds/idl/metaclass_generator.cpp

```diff
@@ -338,6 +338,26 @@ namespace {
     }
     be_global->impl_ << "    }\n";
   }
+
+  void gen_isDcpsKey(IDL_GlobalData::DCPS_Data_Type_Info* info)
+  {
+    if (info && info->key_list_.size()) {
+      for (
```

Comment:
The use of the key_list_ in keys_generator.cpp:74 looks simpler, see if that style would work here too.